File: JXPagerViewExample-OC/JXPagerViewExample-OC/Example/Common/PagingViewTableHeaderView.m
//
//  PagingViewTableHeaderView.m
//  JXCategoryView
//
//  Created by jiaxin on 2018/8/27.
//  Copyright © 2018年 jiaxin. All rights reserved.
//

#import "PagingViewTableHeaderView.h"

@interface PagingViewTableHeaderView()
@property (nonatomic, strong) UIImageView *imageView;
@property (nonatomic, assign) CGRect imageViewFrame;
@property (nonatomic, strong) UILabel *nickLabel;
@end

@implementation PagingViewTableHeaderView

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        _imageView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"lufei.jpg"]];
        self.imageView.clipsToBounds = YES;
        self.imageView.frame = CGRectMake(0, 0, frame.size.width, frame.size.height);
        self.imageView.contentMode = UIViewContentModeScaleAspectFill;
        [self addSubview:self.imageView];
        self.imageView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;

        _nickLabel = [[UILabel alloc] init];
        self.nickLabel.font = [UIFont systemFontOfSize:20];
        self.nickLabel.text = @"Monkey·D·路飞";
        self.nickLabel.textColor = [UIColor redColor];
        [self addSubview:self.nickLabel];
    }
    return self;
}

- (void)layoutSubviews {
    [super layoutSubviews];

    self.imageViewFrame = self.bounds;
    self.nickLabel.frame = CGRectMake(10, self.bounds.size.height - 30, 200, 20);
}

- (void)scrollViewDidScroll:(CGFloat)contentOffsetY {
    CGRect frame = self.imageViewFrame;
    frame.size.height -= contentOffsetY;
    frame.origin.y = contentOffsetY;
    self.imageView.frame = frame;
}


@end

File: Examples/JXPagingViewExample/JXPagingViewExample/Example/Common/PagingViewTableHeaderView.swift
//
//  PagingViewTableHeaderView.swift
//  JXPagingView
//
//  Created by jiaxin on 2018/5/28.
//  Copyright © 2018年 jiaxin. All rights reserved.
//

import UIKit

class PagingViewTableHeaderView: UIView {
    lazy var imageView: UIImageView = UIImageView(image: UIImage(named: "lufei.jpg"))
    var imageViewFrame: CGRect = CGRect.zero

    override init(frame: CGRect) {
        super.init(frame: frame)

        imageView.clipsToBounds = true
        imageView.contentMode = .scaleAspectFill
        imageView.frame = CGRect(x: 0, y: 0, width: frame.size.width, height: frame.size.height)
        imageView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
        self.addSubview(imageView)

        let label = UILabel(frame: CGRect(x: 10, y: frame.size.height - 30, width: 200, height: 30))
        label.font = UIFont.systemFont(ofSize: 20)
        label.text = "Monkey·D·路飞"
        label.textColor = UIColor.red
        label.autoresizingMask = [.flexibleRightMargin, .flexibleTopMargin]
        self.addSubview(label)

        let follow = UIButton(type: .system)
        follow.setTitle("关注", for: .normal)
        follow.addTarget(self, action: #selector(followDidClick), for: .touchUpInside)
        follow.frame = CGRect(x: label.frame.maxX + 10, y: label.frame.minY, width: 50, height: 30)
        addSubview(follow)
        follow.translatesAutoresizingMaskIntoConstraints = false
        if #available(iOS 9.0, *) {
            follow.leadingAnchor.constraint(equalTo: label.trailingAnchor, constant: 5).isActive = true
            follow.centerYAnchor.constraint(equalTo: label.centerYAnchor).isActive = true
        }
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    override func layoutSubviews() {
        super.layoutSubviews()

        imageViewFrame = bounds
    }

    func scrollViewDidScroll(contentOffsetY: CGFloat) {
        var frame = imageViewFrame
        frame.size.height -= contentOffsetY
        frame.origin.y = contentOffsetY
        imageView.frame = frame
    }

    @objc func followDidClick() {
        print("关注成功")
    }

}
